### PR TITLE
Replace memset in saibcm-modules

### PR DESCRIPTION
**Pull request by maipbui**
#### Why I did it
Remove unsafe function - `void *memset( void *dest, int ch, size_t count );` 
- to avoid buffer overflow when `count > sizeof(dest)` and avoid the behavior is undefined when `dest` is a null pointer
- `memset` may be optimized away if the object modified by this function is not accessed again for the rest of its lifetime.
#### How I did it
Replace `memset()` function by `memset_s()` function
`memset_s(): errno_t memset_s( void *dest, rsize_t destsz, int ch, rsize_t count );`
- the errors are detected at runtime and call the currently installed constraint handler function after storing `ch` in every location of the destination range `[dest, dest+destsz)` if buffer overflow occurs.
- This optimization is prohibited for `memset_s`: it is guaranteed to perform the memory write. 
#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205

### CODE REVIEW

File: platform/broadcom/saibcm-modules/sdklt/linux/bde/ngbde_dma.c [hunk 1]
```diff
@@ -224,7 +227,9 @@ ngbde_dmamem_free(ngbde_dmamem_t *dmamem)
         }
         dma_free_coherent(dmamem->dev, dmamem->size,
                           dmamem->vaddr, dmamem->paddr);
-        memset(dmamem, 0, sizeof(*dmamem));
+        #ifdef __STDC_LIB_EXT1__
+            memset_s(*dmamem, sizeof(*dmamem), 0, sizeof(*dmamem));
+	#endif
```
Comment (qiluo-msft) [inline platform/broadcom/saibcm-modules/sdklt/linux/bde/ngbde_dma.c:232]: >	 [](http://example.com/codeflow?start=0&length=1)

Do not mix using tabs and spaces for indentation. Seems this file is using 4-space as indentation.

File: platform/broadcom/saibcm-modules/sdklt/linux/bde/ngbde_dma.c [hunk 2]
```diff
@@ -224,7 +227,9 @@ ngbde_dmamem_free(ngbde_dmamem_t *dmamem)
         }
         dma_free_coherent(dmamem->dev, dmamem->size,
                           dmamem->vaddr, dmamem->paddr);
-        memset(dmamem, 0, sizeof(*dmamem));
+        #ifdef __STDC_LIB_EXT1__
```
Comment (qiluo-msft) [inline platform/broadcom/saibcm-modules/sdklt/linux/bde/ngbde_dma.c:230]: >__STDC_LIB_EXT1__ [](http://example.com/codeflow?start=15&length=17)

If the macro is not defined, you need to fallback to original function.

File: platform/broadcom/saibcm-modules/sdklt/linux/bde/ngbde_dma.c [hunk 3]
```diff
@@ -35,6 +35,9 @@ MODULE_PARM_DESC(dma_debug,
 /*! Default number of DMA memory pools per device. */
 #define NUM_DMAPOOL_DEFAULT     1
 
+/*! Define the macro to the integer constant. */
```
Comment (qiluo-msft) [inline platform/broadcom/saibcm-modules/sdklt/linux/bde/ngbde_dma.c:38]: >Define the macro to the integer constant [](http://example.com/codeflow?start=4&length=40)

The comment does not help reader. You may remove or refine the words.

File: platform/broadcom/saibcm-modules/sdklt/linux/bde/ngbde_dma.c (hunk 1 above)
Comment (maipbui) [reply platform/broadcom/saibcm-modules/sdklt/linux/bde/ngbde_dma.c]: fixed.

File: platform/broadcom/saibcm-modules/sdklt/linux/bde/ngbde_dma.c (hunk 2 above)
Comment (maipbui) [reply platform/broadcom/saibcm-modules/sdklt/linux/bde/ngbde_dma.c:228]: Make senses. Updated fallback function.

File: platform/broadcom/saibcm-modules/sdklt/linux/bde/ngbde_dma.c [hunk 4]
```diff
@@ -35,6 +35,9 @@ MODULE_PARM_DESC(dma_debug,
...
+#define __STDC_WANT_LIB_EXT1__  1
```
[the remark this answers is not in the document]
Comment (maipbui) [reply platform/broadcom/saibcm-modules/sdklt/linux/bde/ngbde_dma.c]: Fixed

File: platform/broadcom/saibcm-modules/sdklt/linux/bde/ngbde_dma.c (hunk 3 above)
Comment (maipbui) [reply platform/broadcom/saibcm-modules/sdklt/linux/bde/ngbde_dma.c]: Done.

File: platform/broadcom/saibcm-modules/sdklt/linux/bde/ngbde_dma.c (hunk 4 above)
[the remark this answers is not in the document]
Comment (qiluo-msft) [reply platform/broadcom/saibcm-modules/sdklt/linux/bde/ngbde_dma.c]: This comment is still applicable.

[the remark this answers is not in the document]
Comment (maipbui) [reply platform/broadcom/saibcm-modules/sdklt/linux/bde/ngbde_dma.c]: Fixed. Please review again.

**Review by qiluo-msft — APPROVED** [review]

**Pull request closed without merging**